src/lib.rs

```diff
@@ -346,172 +191,66 @@ pub trait PolynomialCommitment<F: Field>: Sized {
-    fn batch_open_individual_opening_challenges<'a>(
+    fn multi_point_multi_poly_open<'a>(
         ck: &Self::CommitterKey,
-        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
+        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        query_set: &QuerySet<F>,
+        query_set: &QuerySet<G::ScalarField>,
         fs_rng: &mut Self::RandomOracle,
         rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
         rng: Option<&mut dyn RngCore>,
-    ) -> Result<Self::BatchProof, Self::Error>
+    ) -> Result<Self::MultiPointProof, Self::Error>
         where
             Self::Randomness: 'a,
             Self::Commitment: 'a;
 
-    /// Single point multi poly verify, with random oracle passed from 'outside'.
-    /// CAUTION: This is a low-level function to be handled carefully, typically
-    /// presuming that commitments and query_set is already bound to the internal
-    /// state of the Fiat-Shamir rng.
-    /// TODO: rename this function
-    fn check_individual_opening_challenges<'a>(
+    /// Single point single poly verify:
+    /// Verifies that `value` is the evaluation at `point` of the polynomial
+    /// committed inside `commitment`.
+    fn verify<'a>(
         vk: &Self::VerifierKey,
-        commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        point: F,
-        values: impl IntoIterator<Item = F>,
+        commitment: &'a LabeledCommitment<Self::Commitment>,
+        point: G::ScalarField,
+        value: G::ScalarField,
         proof: &Self::Proof,
+        // This implementation assumes that the commitments, point and evaluations are already absorbed by the Fiat Shamir rng
         fs_rng: &mut Self::RandomOracle,
     ) -> Result<bool, Self::Error>
         where
             Self::Commitment: 'a;
 
-    /// Multi point multi poly verify, with random oracle passed from 'outside'.
+
+    /// Single point multi poly verify, with random oracle passed from 'outside'.
     /// CAUTION: This is a low-level function to be handled carefully, typically
     /// presuming that commitments and query_set is already bound to the internal
     /// state of the Fiat-Shamir rng.
     /// TODO: rename this function
-    fn batch_check_individual_opening_challenges<'a>(
+    fn single_point_multi_poly_verify<'a>(
         vk: &Self::VerifierKey,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
-        query_set: &QuerySet<F>,
-        evaluations: &Evaluations<F>,
-        proof: &Self::BatchProof,
+        point: G::ScalarField,
+        values: impl IntoIterator<Item = G::ScalarField>,
+        proof: &Self::Proof,
         fs_rng: &mut Self::RandomOracle,
     ) -> Result<bool, Self::Error>
         where
             Self::Commitment: 'a;
```

Comment:
We decided to put the implementation for this function in the trait.
As it is now the trait of a linear polynomial commitment scheme, that is the right place.

Reply:
Done